Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionCocoa.mm

```diff
@@ -478,11 +466,10 @@
     if (NSString *cachedString = objectForKey<NSString>(m_resources, path))
         return cachedString;
 
-    bool isServiceWorker = backgroundContentIsServiceWorker();
-    if (!isServiceWorker && [path isEqualToString:generatedBackgroundPageFilename])
+    if ([path isEqualToString:generatedBackgroundPageFilename] && !backgroundContentIsServiceWorker())
         return generatedBackgroundContent();
```

Comment:
This can be simplified now that I look at it:
```suggestion
    if ([path isEqualToString:generatedBackgroundPageFilename] || [path isEqualToString:generatedBackgroundServiceWorkerFilename])
        return generatedBackgroundContent();
```
Then delete the second `if`.